### PR TITLE
Your CQC is sloppy ( CQC costs 30 stamina to block, instead of 25% chance to fail)

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

CQC has a 25% to fail any given block, which seems kind of lame for someone who is supposed to be a master of close quarters combat. In the spirit of sleeping carp being totally immune to projectile attacks, I did the same for CQC and melee attacks, with a caveat; blocking costs 30 stamina instead.

You begin to take slowdown at 40 stamina/total damage (10 brute/30 stamina for instance), so if you're totally unharmed, you can block one attack without taking any slowdown for the effort.
## Why It's Good For The Game

If you try to attack a CQC user in melee combat, alone, you should be totally and completely fucked. But, we don't want CQC users to be immune to stun batons or something equally ridiculous, so adding a stamina cost to a successful block seems like a fair trade. If a nukie wants to get stimmed to the tits and remember the basics of close quarters combat, that sounds fucking sick actually. They would still need to account for projectile weapons in this hypothetical power fantasy. This would also be the case for the chef inside their kitchen; if I leap the counter and try to baton the chef, I shouldn't have a 25% chance to succeed at a close quarters attack against a close quarters combat master. I'll just chuck a CLF3 foam in there or something because fuck that.
## Changelog
:cl: Bisar
balance: CQC users have remembered the basics; they no longer have a 25% chance to fail a block, and instead spend 30 stamina for a successful block.
/:cl:

### CODE REVIEW

File: code/datums/martial/cqc.dm [hunk 1]
```diff
@@ -50,15 +50,14 @@
 		return NONE
 	if(attack_type == PROJECTILE_ATTACK)
 		return NONE
-	if(!prob(block_chance))
-		return NONE
```
Comment (MrMelbert) [inline code/datums/martial/cqc.dm:54]: Should probably check that the user has enough stamina to block without being stamcrit, so people don't just floor themselves

**Comment by Loafin34** [comment]
imagining a clown op with e-shield just being perpetually immortal to any attack short of a bomb

**Comment by necromanceranne** [comment]
I very much think this makes the instances in which players encounter CQC blocking more lame than it is now, but that newer players piloting this version of CQC may find the stamina damage a major noob trap that gets them kill if they weren't aware of the mechanic. 

There is no cooldown limitations or anything, so if enough melee/unarmed attacks get thrown their way in a short span of time, they might just find themselves stamcrit even through their armor.

These small adjustments are not going to make the difference in fixing the issues with CQC. The entire thing is broken from the ground up, and it needs a total re-imagining to fix it. I don't think this meaningfully resolves the problem, but create new ones that aggravate existing pain points for those who have to experience CQC firsthand.

>I dont like it being flat damage especially because ranged weapons have a very very high fire rate compared to melee clicks, youd very likely just be gimping yourself and preparing the enemy for an easy one hit stun baton if someone were to have that rapid fire yellow laser gun to set your stam low first.

This is a massive nerf to cqcs anti range and i think id rather just wear armor / use mutations and species and healing to tank laser damage than ever use cqc to block it.

CQC does not do anything against projectiles, and this damage does not trigger on projectile attacks. It only triggers on attacks from adjacent targets, who have hit the user and they perform a stun attack in retaliation. In fact, as a result of it the adjacency need, I think this means the CQC user is immune to thrown weapons.

**Comment by SmArtKar** [comment]
As Anne said, this doesn't do anything about CQC's main strength which is instant aggro grabs, which are pretty much CQC's entire identity. I am not sure if CQC even ***needs*** nerfs to be honest, I believe that it's good as-is. If nukies manage to kill a captain with it, I'll applaud for it because they had the balls to go into unarmed combat. If you really want to nerf it, make CQC blocks and grabs require empty hand(s?). This just turns CQC blocks into a noob trap that will get you killed.

**Comment by timothymtorres** [comment]
> **Edit:** Additional thing to highlight, if it did do the damage when blocking thrown objects, you could stamcrit a nukie through his armor by random objects at him. This also can burst his energy shield if he has one (which is true on live and kind of intended). Let's not let that happen, okay?

This was the first thing I thought and is exactly the type of shit our playerbase would use as a cheese strat. Greytide would toss paper airplanes at CQC to make them go into stamcrit.

> If you really want to nerf it, make CQC blocks and grabs require empty hand(s?), that'll get rid of the desword combo (which I think is dishonorable).

This is pretty good. I swore sleepy carp had empty hands coded as a perquisite to block projectiles but I couldn't find it in the code anywhere.

**Comment by SmArtKar** [comment]
> This is pretty good. I swore sleepy carp had empty hands coded as a perquisite to block projectiles but I couldn't find it in the code anywhere.

twas got removed (by Anne I think?)

**Comment by SmArtKar** [comment]
> That wasn't a thing. The only thing it checks is for hands use AKA are you retrained/disabled in some way, which is what you might be thinking about.
> 
> Do not take this as a prompt to then have it check for free hands, Tim. It'd become a UX nightmare.
> 
> It's probably more constructive to eliminate blocker stacking altogether. Desword stacked with other blockers is not exclusive to CQC, though it is definitely a good example.

Block code is abyssmal, we'd need to rewrite it entirely (we probably should) to make it work without it being horrifically jank. Kapu did it on daedalus, could look into his PR?

**Comment by Metekillot** [comment]
I have to confess my confusion at anyone thinking this is meant to be a nerf. I also want to express frustration at people arguing against an imaginary CQC that interacted with ranges objects.

The goal is to remove a % chance to fail a close combat maneuver against a close combat master. The stamina cost is meant to be a balance against nukies or iron chefs becoming totally immune to stun batons.

CQC only blocks when in throw mode.

**Comment by Sk1Wass3r** [comment]
I'm not here to argue anything about balance, but rng based combat in general is MASSIVELY UNFUN imo (see sunrise deathmatch, fighting people in melee where everyone has 50% block chance), so i think this is a step in the right direction
(PS PS what are people crying about I've seen like 3 nukies ever actually use CQCs throw mode blocks)

**Comment by PowerfulBacon** [comment]
You could experiment with giving the blocking a cooldown instead of a stamina cost, on bee we have all of our blocking based on stamina and no probability at all and it really sucks. There are 2 reasons it sucks, one of which doesn't apply here:
- You can't turn off blocking without going into help intent which means players can get their arms disabled if they involuntarilly block too much
- If someone is blocking you can never attack them which makes it extremely unfair to play against.

**Pull request closed without merging**